### PR TITLE
Fix the flakiness of the Event_FireList test by using blocking queries

### PR DESCRIPTION
**Pull request by marcin-krystianc**
Fixes the flakiness of the `Event_FireList` by usage of Consul blocking queries feature.

### CODE REVIEW

File: Consul.Test/EventTest.cs [hunk 1]
```diff
@@ -45,21 +45,15 @@ public void Dispose()
...
-            await Task.Delay(100);
-
+            var previousIndex = (await _client.Event.List()).LastIndex;
+            var userEvent = new UserEvent() {Name = "foo"};
+            var res = await _client.Event.Fire(userEvent);
             Assert.NotEqual(TimeSpan.Zero, res.RequestTime);
             Assert.False(string.IsNullOrEmpty(res.Response));
 
-            var events = await _client.Event.List();
+            var events = await _client.Event.List("foo", new QueryOptions {WaitIndex = previousIndex});
```
Comment (jgiannuzzi) [inline Consul.Test/EventTest.cs:54]: I would use `userEvent.Name` instead of `"foo"` here

File: Consul.Test/EventTest.cs [hunk 2]
```diff
@@ -45,21 +45,15 @@ public void Dispose()
...
         public async Task Event_FireList()
         {
-            var userevent = new UserEvent()
-            {
-                Name = "foo"
-            };
-
-            var res = await _client.Event.Fire(userevent);
-
-            await Task.Delay(100);
```
Comment (marcin-krystianc) [inline Consul.Test/EventTest.cs:55]: We don't need any delay on client side because we use `WaitIndex` (https://www.consul.io/api/features/blocking).

**Review by jgiannuzzi — COMMENTED** [review]
I _think_ this should be wrapped in `TimeoutUtils.WithTimeout` so we don't wait forever in case of issues.

**Comment by marcin-krystianc** [comment]
> _think_ this should be wrapped in `TimeoutUtils.WithTimeout` so we don't wait forever in case of issues.
> 
> Edit: though maybe using `WaitTime` in the query options would be enough? Maybe we should do both?

If `WaitTime` is not set then the default timeout is used which according to my tests is 10 min (it is definitely not forever 😄 ).
I think it is better to avoid adding any extra timeout logic (like `TimeoutUtils.WithTimeout`) if it is not absolutely necessary, I believe it is good as it is now.

**Review by jgiannuzzi — APPROVED** [review]
Alright, approved then!

Thanks @marcin-krystianc 👍

**Pull request merged**